common-util/api/leaderboard.ts

```diff
@@ -15,82 +9,59 @@ const CERAMIC_OBJECT = new CeramicClient(API_URL);
 export const getLeaderboardList = async () => {
   const response = await fetch('/api/leaderboard');
   const json: ContributeAgent[] = await response.json();
-  const usersList: ContributeAgent['json_value'][] = [];
+  const usersList: LeaderboardUser[] = [];
 
   if (json && Array.isArray(json)) {
     json.forEach((user) => {
       if (!user.json_value.wallet_address) return;
       if (user.json_value.points === 0) return;
-      usersList.push(user.json_value);
+      usersList.push({
+        ...user.json_value,
+        rank: null,
+        attribute_id: user.attribute_id,
+      });
     });
   }
 
   return usersList;
 };
 
-export const updateUserStakingData = async (
-  twitterId: string | null,
-  multisig: string,
-  serviceId: string,
-) => {
-  const provider = new Ed25519Provider(
-    fromString(process.env.NEXT_PUBLIC_CERAMIC_SEED as string, 'base16'),
-  );
-  const did = new DID({ provider, resolver: getResolver() });
-  // Authenticate the DID with the provider
-  await did.authenticate();
-  // The Ceramic client can create and update streams using the authenticated DID
-  CERAMIC_OBJECT.did = did;
-
-  const response = await TileDocument.load<{
-    users: UsersDbContent;
-  }>(CERAMIC_OBJECT, process.env.NEXT_PUBLIC_STREAM_ID as string);
+type UpdateUserStakingDataParams = {
+  attributeId: number;
+  multisig: string;
+  serviceId: number;
+};
 
-  const newContent = cloneDeep<{
-    users: UsersDbContent;
-  }>(response.content);
+export const updateUserStakingData = async ({
```

Comment:
let's use try/catch?

Reply:
I'm not sure it's needed here, the only thing that can fail inside this function is fetch function itself, we use try catch in api endpoint itself and return non-200 response